projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts

```diff
@@ -47,6 +63,66 @@ type jobRequestResult = {
   isSuccessful: boolean;
 };
 
+type TaskStatusResult = {
+  /**
+   * Type of the task
+   */
+  task_type: string;
+  /**
+   * Status of the task
+   */
+  status: string;
+  /**
+   * Result message of the task if no errors occurred
+   */
+  message: string | IJobConvertToNotebookMessage;
+  /**
+   * Error message
+   */
+  error: string;
+};
+
+const pollForTaskCompletion = async (
+  taskId: string,
+  maxAttempts = 43200,
```

Comment:
These don't seem like good defaults

Reply:
My suggestion here was based on the idea that we got a limit of 12 hours for a job to run. That's why I decided it would make sense to initially try to poll every 10 seconds for these 12 hours which would result in 4320 polls. (added one more 0 here it seems.)

This is initial idea but maybe exponential backoff IMO would fit good here.

Reply:
Ok, that reasoning is probably ok for a first iteration. 
But you need to add a comment here explaining that.